lms/djangoapps/courseware/views/views.py

```diff
@@ -330,22 +281,18 @@ def course_info(request, course_id):
             # to access CCX redirect him to dashboard.
             return redirect(reverse('dashboard'))
 
+        # Redirect the user if they are not yet allowed to view this course
+        redirect_if_needed_to_access_course(request, course)
```

Comment:
The other redirects all seemed to be unique to the course info page, so I didn't refactor them.  We can save those for another fun PR when we merge the two pages. :-)